### ISSUE TITLE
Questions about NeighborSampler

### ISSUE THREAD

**Issue by MrShouxingMa** [pyg-team/pytorch_geometric#3373]
## ❓ Questions & Help

Hi @rusty1s , I have two questions about NeighborSampler.
I need to perform graph convolutional neural networks on a large graph. I used the NeighborSampler method you recommended earlier.  The NeighborSampler method completes the graph convolution of the large graph by dividing into small graphs. 
Question 1:
Now, I want to implement standardized GCNs through the NeighborSampler method, but when I try to add the weight coefficient of the GCNs, it does not change after the loss is backpropagated.
I also tried to replace the GCN weight parameter by adding a linear layer or modify the position of linear transformation (for example, before or after using the NeighborSampler method), but it doesn't work!
After I modify, I print the variable：
`Parameter containing:
tensor([[0.3245]], device='cuda:0', requires_grad=True)`
Although it shows that the gradient can be backpropagated, it does not do so. I don't know how to make GCN parameters back-propagating gradients.
Question 2:
There is only one trainable parameter matrix in GCNs. I want to change the adjacency matrix into a trainable parameter matrix at the same time (because Kipf's paper mentions that it can be binary or weighted). Besides the method of using GAT, is there other methods?
Many thanks in advance and thanks for sharing work!
Best regards!
